src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -32,45 +39,31 @@ private class LocalFuncUsages
         {
             _usedLocalFunctions.Add(localFunc);
 
+            var usages = GetOrCreateLocalFuncUsages(localFunc);
+
             // First process the reads
-            ReplayVarUsage(localFunc,
-                           syntax,
-                           isWrite: false);
+            ReplayReads(ref usages.ReadVars, syntax);
 
             // Now the writes
             if (writes)
             {
-                ReplayVarUsage(localFunc,
-                               syntax,
-                               isWrite: true);
+                UnionWith(ref this.State, ref usages.WrittenVars);
```

Comment:
I don't think this will interact properly with the `DataFlowsOutWalker`. See the method `AssignImpl` in that type. The problem probably exists in the previous version of this code. The symptom of that is that the flow analysis API DataFlowsOut will not recognize that an assignment inside the region performed by called a local function "flows out" to a use site after the region. The user-visible symptom would be that extract method won't do the right thing.

This is probably OK for now but I'll file an issue for DataFlowsOut.